content/posts/event-driven-scaling-fission-function-using-keda.md

```diff
@@ -0,0 +1,239 @@
+}
+```
+
+Let’s create the environment and function
+```
+$ fission environment create --name go --image fission/go-env-1.12:1.10.0 --builder fission/go-builder-1.12:1.10.0
+$ fission fn create --name producer --env go --src producer.go --entrypoint Handler
+```
+
+### Creating Consumer Function
+
+The consumer function is a Node Js program. Save the following code with consumer.js
+
+```
+module.exports = async function (context) {
+    console.log(context.request.body);
+    let obj = context.request.body;
+    return {
+        status: 200,
+        body: "Consumer Response "+ JSON.stringify(obj)
+    };
+}
+```
+
+Let’s create the environment and function
+
+```
+$ fission env create --name nodeenv --image fission/node-env
+$ fission fn create --name consumer --env nodeenv --code consumer.js
+```
+
+## Creating Message Queue Trigger
+```
+$ fission mqt create --name kafkatest --function consumer --mqtype kafka --mqtkind keda --topic request-topic --resptopic response-topic --errortopic error-topic --maxretries 3 --metadata bootstrapServers=my-cluster-kafka-brokers.my-kafka-project.svc:9092 --metadata consumerGroup=my-group --metadata topic=request-topic  --cooldownperiod=30 --pollinginterval=5`
+```
+To take leverage advantage of KEDA scaling, one must set `--mqtkind keda`
+
+## Everything in Action
+
+Open one terminal and execute the following to keep watch on pods
+```
+$ kubectl get pods -w
+```
+Let’s subscribe to our response-topic to see the response return by consumer function invocations. Open another terminal and execute the following:
+```
+$ kubectl run kafka-consumer -ti --image=strimzi/kafka:latest-kafka-2.5.0 --rm=true --restart=Never -- bin/kafka-console-consumer.sh --bootstrap-server my-cluster-kafka-brokers.my-kafka-project.svc:9092 --topic response-topic
+```
+
+Since everything is in place, let’s invoke our producer function in another terminal
+```
+$ fission fn test --name producer
+```
+
+Viola! Now you can see eventually three new pods are being created in the first terminal and response of the consumer function in the second terminal. 
+
+You would probably have an output like this in the first terminal
+
+![consumer pods](../../../images/event-driven-scaling-fission-function-using-keda/pods_watch.png)
+
+And output like this in the second terminal
+
+![response messages](../../../images/event-driven-scaling-fission-function-using-keda/response.png)
+
+# What next?
+
+We have developed the Keda Integration and Kafka connector so far and we soon will be adding connectors for RabbitMQ, NATS too. We would love to see more connectors being contributed by the community as there are many integrations that Keda supports!
+
+![scalers](../../../images/event-driven-scaling-fission-function-using-keda/scalers.png)
+
+
+--- 
+
+
+**_Authors:_**
+
+* [Vishal Biyani](https://twitter.com/vishal_biyani)  **|**  [Fission Contributor](https://github.com/vishal-biyani)  **|**  CTO - [InfraCloud Technologies](http://infracloud.io/)
```

Comment:
You can please remove my name - and only keep your name - this is all your work!

Reply:
ok